Locker staff fixes and tweaks

MNarath1 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This PR fixes known bugs with the locker staff like not putting mobs into locker at all as well as lockers that did not hit any mobs beeing useable to infinitly produce iron. This also includes some tweaks to the lockerstaff first of all there is a cap of 5 people for the lockerstaff now (trough this will be enforced at the bolt scooping people up and not at the locker) so that people above the lockercap are not just spawing ontop of the locker after it hits except those 3 poor random people that got stuck inside buckled people can also be sucked in either buckled on a chair or a vehicle or on a borg. Also lockers that spawn and did not scoop up any mobs will vanish after 18 seconds.
Also small refractor of locker staff code i guess.
closes #2516
## Why It's Good For The Game

Finally fixing lockerstaff and making it actually a usefull item for wizards again

## Changelog
:cl:
tweak: up to 5 mobs can end up in a cursed locker now
tweak: Lockers that did not scoop up any mobs vanish after 18 seconds
tweak: buckled mobs can be scooped up too(not AI)
fix: lockerstaff now actually putting people into lockers
fix: another infinit iron dublication glitch
refactor: deleting some unneeded code
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

MNarath1 committed:
fix

MNarath1 committed:
removes unneeded checks

MNarath1 committed:
change

MNarath1 commented:
Well if anyone has suggestions balance wise i would be open for them cause while i think this is needed to fix the staff i do also think this thing is horrible op i mean seriously it locks you into that locker for 5 minutes straight.

MNarath1 committed:
capping the lockerstaff bolt to a content of 5 people

that way not half the station ends up in a single  locker  but its still enough to get you sec or a crowd of people off you

MNarath1 commented:
basically with the last commit the locker bolt will stop when it hits a mob after reaching its capacity limit (without scooping them up) i moved the content cap check from the locker to the bolt to avoid people getting dragged along it and then just ending up above the locker.
Note trough cause only living mobs can be dragged along by the bolt this does not need to check if the contents are actually mobs so it can just get the length of the list

PowerfulBacon commented on code/modules/projectiles/projectile/magic.dm:
```diff
@@ -430,6 +426,7 @@
 
 /obj/structure/closet/decay/proc/magicly_lock()
 	if(!welded)
+		addtimer(CALLBACK(src, .proc/decay), 15 SECONDS)
```
is there a point for auto_destroy now if they decay automatically?

MNarath1 replied:
Reason for why i did it was cause i felt that 5 minutes at least for the lockers that did not catch anything is too long trough i give in this is a bad way to implement it i will look into making sure only one timer runs at a time (so that there is not a totally obsolete timer that will not run out in time anyway.

MNarath1 replied:
I will just use a switch expression to choose what timers to add

MNarath1 replied:
Ok so much about using a switch or if in initialize sadly i still had to make the proc call before the check

MNarath1 committed:
Why

This is not ideal i tried to put the check in Initialize but that came to be impossible for reasons(seems like the welded = TRUE gets applied after Initialize so yeah instead i put it into magicallock so people are stuck for 5 deciseconds longer than before

MNarath1 commented:
I honestly do not like how i had to solve this here but i cannot check it in initialize sadly also no idea why the original coder of this made the auto_destroy var anyways its not like there could be a situation where it would end up with the auto_destroy var beeing false

MNarath1 committed:
replace the if(!welded) return with if(welded) else

Given that i already moved the addtimer to this proc i think this is way better then if(!welded) return

MNarath1 commented:
I think using if else is better then using if return in this case

MNarath1 committed:
rename proc

I decided to rename this proc cause i moved timer in there to actually be able to check for for welded or not welded cause calling the check in initialize won't work because c.welded = TRUE had no time to actually override the welded var yet

ike709 commented:
I think this looks fine but I'll request another review to be sure.

utilbird commented on code/modules/projectiles/projectile/magic.dm:
```diff
@@ -391,7 +386,7 @@
 			qdel(src)
 			return
-		if(!locker_temp_instance.insertion_allowed(M))
+		if(!isliving(M) || M.incorporeal_move || M.mob_size > MOB_SIZE_HUMAN || LAZYLEN(contents)>=5)
```
There's already a living check above

MNarath1 committed:
deleted a double check

co-authored-by: TheFakeElon <TheFakeElon@user.noreply.github.com>

utilbird commented:
@ike709 looks good
